src/Xamarin.Android.Build.Tasks/Tasks/GetMonoPlatformJar.cs

```diff
@@ -2,42 +2,51 @@
 
 using System;
 using System.IO;
+using System.Linq;
+using System.Collections.Generic;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 
 namespace Xamarin.Android.Tasks
 {
-	public class GetMonoPlatformJar : Task
+	public class GetMonoPlatformJars : Task
 	{
 		[Required]
 		public string TargetFrameworkDirectory { get; set; }
 
+		[Required]
+		public string[] JarFiles { get; set; }
+
 		[Output]
-		public string MonoPlatformJarPath { get; set; }
+		public ITaskItem[] MonoPlatformJarPaths { get; set; }
 
 		[Output]
-		public string MonoPlatformDexPath { get; set; }
+		public ITaskItem[] MonoPlatformDexPaths { get; set; }
 
 		public override bool Execute ()
-		{
-			Log.LogDebugMessage ("GetMonoPlatformJar Task");
-			Log.LogDebugMessage ("  TargetFrameworkDirectory: {0}", TargetFrameworkDirectory);
-			
+		{			
 			var directories = TargetFrameworkDirectory.Split (new char[] { ';', ','} ,StringSplitOptions.RemoveEmptyEntries);
-
+			List<string> foundJars = new List<string> ();
+			List<string> foundDexes = new List<string> ();
 			foreach (var dir in directories)
-				if (File.Exists (Path.Combine (dir, "mono.android.jar"))) {
-					MonoPlatformJarPath = Path.Combine (dir, "mono.android.jar");
-					MonoPlatformDexPath = Path.ChangeExtension (MonoPlatformJarPath, ".dex");
-					Log.LogMessage (MessageImportance.Low, "  MonoPlatformJarPath: {0}", MonoPlatformJarPath);
-					Log.LogMessage (MessageImportance.Low, "  MonoPlatformDexPath: {0}", MonoPlatformDexPath);
-
-					return true;
+				foreach (var jar in JarFiles) {
+					if (File.Exists (Path.Combine (dir, jar))) {
+						var monoPlatformJarPath = Path.Combine (dir, jar);
+						var monoPlatformDexPath = Path.ChangeExtension (monoPlatformJarPath, ".dex");
+						foundJars.Add (monoPlatformJarPath);
+						foundDexes.Add (monoPlatformDexPath);
+					}
 				}
 
-			Log.LogError ("Could not find mono.android.jar");
+			if (foundJars.Count == 0) {
+				Log.LogError ("Could not find mono.android.jar");
```

Comment:
Should this actually be an error?

If it is, do we need a proper `XA....` error code and corresponding documentation for this new error?  Or should we use `XA0000`?